src/hooks/useLearntQuestions/useLearntQuestions.utils.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  Question,
+  Subject,
+  LearntQuestions,
+  learntQuestionSchema,
+} from 'validators/subjects';
+
+type SubjectId = Subject['id'];
+export type QuestionId = Question['id'];
+
+type LearntQuestionsKey = `learnt-${string}`;
+
+export const getLearntQuestionsLocalStorageKey = (
+  subjectId: SubjectId,
+): LearntQuestionsKey => `learnt-${subjectId}`;
+
+export const getLearntQuestionsFromLocalStorage = (
+  key: LearntQuestionsKey,
+): LearntQuestions => {
+  const questions = localStorage.getItem(key);
+
+  if (!questions) {
+    return [];
+  }
+
+  return learntQuestionSchema.validateSync(JSON.parse(questions));
```

Comment:
Let's also `try/catch` this (both for the `JSON.parse` and the `validateSync`) and return a default value in case of an error